File: tasks/3545-minimum-deletions-for-at-most-k-distinct-characters.ts
// Accepted 936 / 936 testcases passed Sergey Pomortsev submitted at Mar 04, 2026 18:55
// Solution Runtime 0 ms Beats 100.00% Analyze Complexity Memory 58.84 MB Beats 14.29%

const minDeletion = (s: string, k: number): number => {
  const map = new Map();

  for (let i = 0; i < s.length; i++) {
    map.set(s[i], map.has(s[i]) ? map.get(s[i]) + 1 : 1);
  }

  return Array.from(map.values()).sort((a, b) => b - a).slice(k).reduce((acc, curr) => acc + curr, 0);
}

File: tasks/3545-minimum-deletions-for-at-most-k-distinct-characters.go
// Accepted 936 / 936 testcases passed Sergey Pomortsev submitted at Jul 12, 2026 11:55
// Solution Runtime 0 ms Beats 100.00% Memory 4.41 MB Beats 0.00%

func minDeletion(s string, k int) int {
  delCount := 0
  freq := make([]int, 26)

  for _, r := range s {
    freq[r - 'a']++
  }

  sort.Slice(freq, func(a, b int) bool {
    return freq[a] > freq[b]
  })

  for i := k; i < len(freq); i++ {
    delCount += freq[i]
  }

  return delCount
}